File: backend/springnatour/src/main/java/com/unina/springnatour/dto/chat/ChatDto.java
package com.unina.springnatour.dto.chat;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.unina.springnatour.dto.user.UserDto;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDate;

@Data
public class ChatDto implements Serializable {
    @JsonProperty("chat_id")
    private Long chatId;

    @JsonProperty("chat_creation_date")
    private LocalDate chatCreationDate;

    @JsonProperty("chat_user_1")
    private UserDto chatUser1;

    @JsonProperty("chat_user_2")
    private UserDto chatUser2;
}

File: client/app/src/main/java/com/unina/natourkt/core/data/remote/dto/ChatDto.kt
package com.unina.natourkt.core.data.remote.dto

import com.google.gson.annotations.SerializedName

data class ChatDto(
    @SerializedName("chat_id")
    val chatId: Long,

    @SerializedName("chat_creation_date")
    val chatCreationDate: String,

    @SerializedName("chat_user_1")
    val chatUser1: UserDto,

    @SerializedName("chat_user_2")
    val chatUser2: UserDto,
)